examples_notebooks/community_contrib/couchbase/GraphRAG_with_Couchbase.ipynb

```diff
@@ -0,0 +1,557 @@
+    "        api_key=OPENAI_API_KEY,\n",
+    "        model=LLM_MODEL,\n",
+    "        api_type=OpenaiApiType.OpenAI,\n",
+    "        max_retries=20,\n",
+    "    )\n",
+    "\n",
+    "    token_encoder = tiktoken.get_encoding(\"cl100k_base\")\n",
+    "\n",
+    "    text_embedder = OpenAIEmbedding(\n",
+    "        api_key=OPENAI_API_KEY,\n",
+    "        api_base=None,\n",
+    "        api_type=OpenaiApiType.OpenAI,\n",
+    "        model=EMBEDDING_MODEL,\n",
+    "        deployment_name=EMBEDDING_MODEL,\n",
+    "        max_retries=20,\n",
+    "    )\n",
+    "\n",
+    "    logger.info(\"LLM and embedding models setup completed\")\n",
+    "except Exception as e:\n",
+    "    logger.error(f\"Error setting up models: {str(e)}\")\n",
+    "    raise"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "r08V2NuugdSF"
+   },
+   "source": [
+    "# Storing Embeddings in Couchbase\n",
+    "After generating embeddings for the entities, we store them in Couchbase. We use the store_entity_semantic_embeddings function to store the embeddings.\n",
+    "\n",
+    "This method checks if the input is either a dictionary or a list and processes it accordingly.\n",
+    "It uses the Couchbase vector store to save the embeddings, ensuring that entities have the proper 'id' attribute for storage.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "id": "C1wV0RqrgdVl"
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024-09-06 15:34:49,609 - __main__ - INFO - Storing entity embeddings\n",
+      "2024-09-06 15:34:49,612 - graphrag.vector_stores.couchbasedb - INFO - Loading 96 documents into vector storage\n",
+      "2024-09-06 15:34:49,998 - graphrag.vector_stores.couchbasedb - INFO - Successfully loaded 96 out of 96 documents\n",
+      "2024-09-06 15:34:50,007 - __main__ - INFO - Entity semantic embeddings stored successfully\n"
+     ]
+    }
+   ],
+   "source": [
+    "logger.info(f\"Storing entity embeddings\")\n",
+    "\n",
+    "try:\n",
+    "    entities_list = list(data[\"entities\"].values()) if isinstance(data[\"entities\"], dict) else data[\"entities\"]\n",
+    "\n",
+    "    store_entity_semantic_embeddings(\n",
+    "        entities=entities_list, vectorstore=description_embedding_store\n",
+    "    )\n",
+    "    logger.info(\"Entity semantic embeddings stored successfully\")\n",
+    "except AttributeError as e:\n",
+    "    logger.error(f\"Error storing entity semantic embeddings: {str(e)}\")\n",
+    "    logger.error(\"Ensure all entities have an 'id' attribute\")\n",
+    "    raise\n",
+    "except Exception as e:\n",
+    "    logger.error(f\"Error storing entity semantic embeddings: {str(e)}\")\n",
+    "    raise"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f0LSSSJlgdZd"
+   },
+   "source": [
+    "# Building the Search Engine\n",
```

Comment:
This whole section can be explained a bit more. Around the lines of what it does and how

Reply:
will explain all the params